src/internal/modules/billing/lib/two-part-tariff.js

```diff
@@ -21,11 +22,40 @@ const getErrorString = errorCodes => errorCodes.reduce((acc, code) => {
   return acc ? 'Multiple errors' : statusMessages.get(code);
 }, null);
 
-const mapLicence = (batch, licence, action) => ({
-  ...licence,
-  twoPartTariffStatuses: getErrorString(licence.twoPartTariffStatuses),
-  link: routing.getTwoPartTariffLicenceReviewRoute(batch, licence.licenceId, action)
-});
+/**
+ * Creates a unique group string for the given billing volume, based on the
+ * purpose and abstraction period of its charge element
+ * @param {Object} billingVolume
+ * @return {String} unique key
+ */
+const getBillingVolumeGroup = billingVolume => {
+  const { chargeElement } = billingVolume;
+  const { startDay, startMonth, endDay, endMonth } = chargeElement.abstractionPeriod;
+  return `${chargeElement.purposeUse.code}_${startDay}_${startMonth}_${endDay}_${endMonth}`;
+};
+
+const mapLicence = (batch, licenceGroup) => {
+  const mappedLicence = licenceGroup.reduce((acc, licence) => {
+    acc.licenceId = licence.licenceId;
+    acc.licenceRef = licence.licenceRef;
+    acc.billingContact = `${licence.billingContact}\n${acc.billingContact}`;
+    acc.twoPartTariffError = acc.twoPartTariffError ? acc.twoPartTariffError : licence.twoPartTariffError;
```

Comment:
If twoPartTariffError and billingVolumeEdited are booleans I think these lines could maybe be shortened to e.g.:
```
acc.twoPartTariffError = acc.twoPartTariffError ? acc.twoPartTariffError : licence.twoPartTariffError;
```